pkg/networking/kindnetd.go

```diff
@@ -0,0 +1,58 @@
+package networking
+
+import (
+	"fmt"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/yaml"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+)
+
+type Kindnetd struct{}
+
+func NewKindnetd() *Kindnetd {
+	return &Kindnetd{}
+}
+
+func (c *Kindnetd) GenerateManifest(clusterSpec *cluster.Spec) ([]byte, error) {
+	content, err := loadManifest(clusterSpec, clusterSpec.VersionsBundle.Kindnetd.Manifest)
+	if err != nil {
+		return nil, fmt.Errorf("can't load kindnetd manifest: %v", err)
+	}
+	var finalContent string
+	templates := strings.Split(string(content), "---")
+	for _, template := range templates {
+		u := &unstructured.Unstructured{}
+		if err := yaml.Unmarshal([]byte(template), u); err != nil {
+			return nil, fmt.Errorf("unmarhsaling kindnetd type [%s]: %v", template, err)
+		}
+		if u.GetKind() == "DaemonSet" {
+			updated, err := updatePodSubnet(clusterSpec, u)
+			if err != nil {
+				return nil, fmt.Errorf("updating kindnetd pod subnet [%s]: %v", template, err)
+			}
+			template = "\n" + strings.TrimSuffix(string(updated), "\n")
+		} else {
+			template += "---"
```

Comment:
nit: I think you could simplify the logic by using the helper `templater.AppendYamlResources`